### PR TITLE
Add some fields to the DB; add disposable event sourcing experiment

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Adds some fields to the DB, per https://github.com/holochain/holochain/issues/4241. 

Also implements the basics of my event sourcing idea. Most of the lines in this PR are adding the new `event` module, which is totally encapsulated and not depended on anywhere else, so if you ignore those files, this PR is actually quite a small change.

The new DB fields pertaining to the FetchPool will be in a followup PR, since that requires a lot of extra plumbing throughout the system.

### TODO:
- [x] CHANGELOGs updated with appropriate info
- [ ] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

File: crates/holochain_sqlite/src/schema.rs [hunk 1]
```diff
@@ -29,6 +29,10 @@ pub static SCHEMA_CELL: Lazy<Schema> = Lazy::new(|| Schema {
             forward: include_str!("sql/cell/schema/2-up.sql").into(),
             _schema: include_str!("sql/cell/schema/2.sql").into(),
         },
+        M {
+            forward: include_str!("sql/cell/schema/3-up.sql").into(),
+            _schema: include_str!("sql/cell/schema/2.sql").into(),
```
Comment (maackle) [inline crates/holochain_sqlite/src/schema.rs:34]: TODO, write 3.sql

File: crates/holochain_state/src/event.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+use crate::prelude::*;
```
Comment (maackle) [inline crates/holochain_state/src/event.rs:1]: This whole module can be removed at any point if you like. This is the part we don't agree on being useful.

File: crates/holochain_types/src/dht_op.rs [hunk 3]
```diff
@@ -647,6 +668,24 @@ impl ChainOp {
         Ok(r)
     }
 
+    /// "Normalize" the op by running it through `from_type` to give a sensisble
```
Comment (jost-s) [inline crates/holochain_types/src/dht_op.rs:671]:
```suggestion
    /// "Normalize" the op by running it through `from_type` to give a sensible
```

File: crates/holochain_state/src/event.rs (hunk 2 above)
Comment (jost-s) [reply crates/holochain_state/src/event.rs:1]: If the usefulness is doubted, I'd tend to exclude this module from the PR.

**Review by jost-s — APPROVED** [review]
LGTM

File: crates/holochain_state/src/event.rs (hunk 2 above)
Comment (maackle) [reply crates/holochain_state/src/event.rs:1]: I personally think it's quite useful, but if everybody else does not, somebody should make a commit to delete it.

**Review by neonphog — APPROVED** [review]
removed event source. lgtm

**Pull request merged**